sound/soc/sof/sof-client-ipc-msg-injector.c

```diff
@@ -311,7 +359,8 @@ static void sof_msg_inject_remove(struct auxiliary_device *auxdev)
 
 	pm_runtime_disable(&auxdev->dev);
 
-	debugfs_remove(priv->dfs_file);
+	debugfs_remove(priv->kernel_dfs_file);
+	debugfs_remove(priv->fw_dfs_file);
```

Comment:
Given that this feature basically adds a 'fake' DSP which sends messages to the kernel, I think it would be better to have a new Kconfig option with a new client driver and have the needed interfaces (sof-clientized!) be #idfef wrapped.

Reply:
That would require taking any common code and moving to a third file, what is the motive behind splitting this from the existing injector?

Reply:
In the 'fake' DSP case you only need the write debugfs file, I don't know what you would do with the read in case you torture the Kernel...